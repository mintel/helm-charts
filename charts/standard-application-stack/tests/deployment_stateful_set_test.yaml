suite: Test Stateful Set
templates:
  - deployment.yaml
release:
  namespace: test-namespace
tests:
  - it: Check stateful set is rendered with volumeClaimTemplates
    set:
      global.clusterEnv: qa
      global.name: test-app
      statefulset: true
      persistentVolumes:
        - name: 'vol1'
          mode: 'ReadWriteOnce'
          size: '10Mi'
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: StatefulSet
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app
      - equal:
          path: spec.volumeClaimTemplates[0].metadata.name
          value: test-app-vol1
      - equal:
          path: spec.volumeClaimTemplates[0].spec
          value:
            accessModes:
            - ReadWriteOnce
            resources:
              requests:
                storage: 10Mi
