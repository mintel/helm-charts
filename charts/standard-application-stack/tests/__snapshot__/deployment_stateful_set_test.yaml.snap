Check stateful set is rendered with volumeClaimTemplates:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      serviceName: test-app
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              matchLabelKeys:
                - pod-template-hash
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
      updateStrategy:
        type: RollingUpdate
      volumeClaimTemplates:
        - metadata:
            name: test-app-vol1
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 10Mi
